### PR TITLE
API operation list

### PR DESCRIPTION
**Pull request by RupengLiu**
Without tag:
image
image
With tag:
image
image
Search with paging:
image
image

### CODE REVIEW

File: src/components/operations/operation-list/ko/runtime/operation-list.ts [hunk 1]
```diff
@@ -109,28 +155,48 @@ export class OperationList {
...
+        if (this.groupByTag()) {
+            this.searchRequest.grouping = "tag";
+        } else {
+            this.searchRequest.grouping = "none";
+        }
+        this.searchOperations();
+    }
+
+    private async loadBytag(): Promise<void> {
+        this.searchRequest.skip = (this.pageNumber() - 1) * Constants.defaultPageSize;
```
Comment (ygrik) [inline src/components/operations/operation-list/ko/runtime/operation-list.ts]: you should reset page number on pattern change

File: src/components/operations/operation-list/ko/runtime/operation-list.ts [hunk 2]
```diff
@@ -109,28 +155,48 @@ export class OperationList {
...
+        const pageOfOperationsByTag = await this.apiService.getOperationsByTags(this.currentApiId, this.searchRequest);
+        const operationGroups = pageOfOperationsByTag.value;
+        this.opsGroups(operationGroups);
+        this.hasPrePage(this.pageNumber() > 1);
+        this.hasNextPage(!!pageOfOperationsByTag.nextLink);
+    }
+
     private async load(): Promise<void> {
+        this.searchRequest.skip = (this.pageNumber() - 1) * Constants.defaultPageSize;
```
Comment (ygrik) [inline src/components/operations/operation-list/ko/runtime/operation-list.ts]: you should reset page number on pattern change

File: src/services/apiService.ts [hunk 3]
```diff
@@ -74,6 +74,62 @@ export class ApiService {
...
+
+        if (searchQuery) {
+            if (searchQuery.tags && searchQuery.tags.length > 0) {
+                const tagFilterEntries = searchQuery.tags.map((tag) => `tag/name eq '${tag}'`);
+                odataFilterEntries.push(`(${tagFilterEntries.join(" or ")})`);
+            }
+
+            if (searchQuery.pattern) {
+                const pattern = Utils.escapeValueForODataFilter(searchQuery.pattern);
+                odataFilterEntries.push(`(contains(operation/name,'${encodeURIComponent(pattern)}'))`);
```
Comment (ygrik) [inline src/services/apiService.ts]: should we search by operation displayName?

Comment (azaslonov) [reply src/services/apiService.ts]: no, this is how operationByTags resource has it right now by some reason.

Comment (RupengLiu) [reply src/services/apiService.ts]: I tried use name, but it doesn't work, displayname works (pics in the description)

**Pull request merged**